pkg/pgmodel/extension.go

```diff
@@ -234,10 +256,30 @@ func getSqlVersion(v semver.Version, extName string) string {
 }
 
 // getNewExtensionVersion returns the highest version allowed by validRange
-func getNewExtensionVersion(availableVersions semver.Versions, validRange semver.Range) (semver.Version, bool) {
+func getNewExtensionVersion(extName string,
+	availableVersions semver.Versions,
+	defaultVersion semver.Version,
+	validRange semver.Range,
+	validCurrentVersion bool,
+	currentVersion semver.Version) (semver.Version, bool) {
 	//sort higher extensions first
 	sort.Sort(sort.Reverse(availableVersions))
```

Comment:
I don't see availableVersion is already sorted. So, shouldn't this be `sort.Sort(sort.Reverse(sort.SliceStabe(callback)))` ? Or a normal sort and then negative for loop.

Reply:
Semver.Versions has Sort.Interface implemented. So this will sort in descending version order.